typescript-generator/index.js

```diff
@@ -106,7 +106,8 @@ function createInterface (type) {
   return code
 
   function required (property) {
-    if (isResponse) return ''
+    // response default to required, unless explicitly marked as optional
+    if (isResponse) return property.required === false ? '?' : ''
```

Comment:
@delvedor had to update this to allow response to mark direct properties as optional